com.hoptodesk.HopToDesk.json

```diff
@@ -0,0 +1,264 @@
+            "cflags" : "-Wno-error",
+            "env" : {
+              "HAVE_VALGRIND_FALSE" : "#",
+              "HAVE_VALGRIND_TRUE" : ""
+            }
+          },
+          "cleanup" : [
+            "/include",
+            "/libexec",
+            "/lib/pkgconfig",
+            "/lib/*.la",
+            "/share/doc",
+            "/share/libdbusmenu",
+            "/share/gtk-doc",
+            "/share/gir-1.0"
+          ]
+        },
+        {
+          "name" : "libindicator",
+          "config-opts" : [
+            "--disable-static",
+            "--disable-tests",
+            "--with-gtk=3"
+          ],
+          "sources" : [
+            {
+              "url" : "https://launchpad.net/libindicator/12.10/12.10.1/+download/libindicator-12.10.1.tar.gz",
+              "sha256" : "b2d2e44c10313d5c9cd60db455d520f80b36dc39562df079a3f29495e8f9447f",
+              "type" : "archive"
+            },
+            {
+              "commands" : [
+                "# FTBFS fix",
+                "sed -e '/LIBINDICATOR_LIBS/ s/\\$LIBM/ $LIBM/' -i configure.ac"
+              ],
+              "type" : "shell"
+            },
+            {
+              "commands" : [
+                "autoreconf -sfi"
+              ],
+              "dest-filename" : "autogen.sh",
+              "type" : "script"
+            }
+          ],
+          "build-options" : {
+            "cflags" : "-Wno-error"
+          },
+          "cleanup" : [
+            "/include",
+            "/libexec",
+            "/lib/pkgconfig",
+            "/lib/*.la",
+            "/share/libindicator",
+            "/share/gtk-doc",
+            "/share/gir-1.0"
+          ]
+        }
+      ]
+    },
+    {
+      "name" : "xdotool",
+      "buildsystem" : "simple",
+      "sources" : [
+        {
+          "url" : "https://github.com/jordansissel/xdotool/releases/download/v3.20211022.1/xdotool-3.20211022.1.tar.gz",
+          "sha256" : "96f0facfde6d78eacad35b91b0f46fecd0b35e474c03e00e30da3fdd345f9ada",
+          "type" : "archive"
+        }
+      ],
+      "build-commands" : [
+        "make -j4 && PREFIX=./build make install",
+        "cp -r ./build/*  /app/"
+      ]
+    },
+    {
+      "name" : "libpam",
+      "buildsystem" : "simple",
+      "sources" : [
+        {
+          "url" : "https://github.com/linux-pam/linux-pam/releases/download/v1.6.1/Linux-PAM-1.6.1.tar.xz",
+          "sha256" : "f8923c740159052d719dbfc2a2f81942d68dd34fcaf61c706a02c9b80feeef8e",
+          "type" : "archive"
+        }
+      ],
+      "build-commands" : [
+        "./configure --prefix=$(pwd)/build",
+        "make -j4 && make install",
+        "cp -r ./build/*  /app/"
+      ]
+    },
+    {
+      "name" : "HopToDesk",
+      "buildsystem" : "simple",
+      "sources" : [
+        {
+          "url": "https://www.hoptodesk.com/hoptodesk.deb",
+          "sha256": "6fb51d3d61fb82124d153a5f1f554acf72d6597152ed867ab17184b673935ee6",
+          "type" : "file"
```

Comment:
This is x86_64 only you are building for aarch64 too.

Reply:
Also why not build from source?

Reply:
Also why is this unversioned? This will break if the sha256 changes.

Reply:
https://docs.flathub.org/docs/for-app-authors/requirements#flathubjson

Reply:
> This is x86_64 only you are building for aarch64 too.

Updated `flathub.json` for buildng on x86_64 only

Reply:
> Also why is this unversioned? This will break if the sha256 changes.

Building rust apps in flatpak is kind of a pain, I could never manage to build it correctly. Therefore, I decided to use a pre-compiled deb for this purpose.

Reply:
Fixed, the workflow uses a versioned file now

Reply:
> Building rust apps in flatpak is kind of a pain, I could never manage to build it correctly. Therefore, I decided to use a pre-compiled deb for this purpose.

Building Rust is perfectlty fine. Many Rust apps are build on flathub.